core/commonTypes.ml

```diff
@@ -199,6 +199,19 @@ module Name = struct
     [@@deriving show]
 end
 
+module ForeignLanguage = struct
+  type t =
+    | JavaScript
+    [@@deriving show]
+
+  let of_string = function
+    | "javascript" -> JavaScript
```

Comment:
Maybe we should normalise the "language strings" such that a user can type `alien JavaScript ...` or `alien jAvAsCrIpT ...` or whatever.